src/log/log.c

```diff
@@ -2227,30 +2226,21 @@ __log_write_internal(WT_SESSION_IMPL *session, WT_ITEM *record, WT_LSN *lsnp,
 		if (free_slot)
 			__wt_log_slot_free(session, myslot.slot);
 	} else if (force) {
-		/*
-		 * If we are going to wait for this slot to get written,
-		 * signal the wrlsn thread.
-		 *
-		 * XXX I've seen times when conditions are NULL.
-		 */
-		if (conn->log_cond != NULL) {
-			__wt_cond_signal(session, conn->log_cond);
-			__wt_yield();
-		} else
-			WT_ERR(__wt_log_force_write(session, 1, NULL));
+		__wt_cond_signal(session, &conn->log_cond);
+		__wt_yield();
```

Comment:
This change makes me nervous - if a call comes through here and ``conn->log_cond`` has not been initialized, I suspect it'd crash, and the existing comment inferred that could happen. Do you think it can't any longer?